### PR TITLE
Data schema tests

### PR DESCRIPTION
**Pull request by Jung0219**
Enhancing config validation script by adding an interactive prompt for the JSON file path when no command-line argument is provided. 
Key changes:

Preserved existing schema validation and semantic checks (e.g., file_inputs, points, head structure).
Improved usability for testing and interactive runs without requiring command-line arguments.
Validates required keys and value types (file_inputs, points, shown_outputs, etc.).

Adds semantic checks:
right_fin, left_fin, spine, and tail must each have at least two points.
head must contain exactly pt1 and pt2, both as strings.
Supports both command-line and interactive usage:
Run with a file path argument: python verify_config.py path/to/config.json
Run without arguments to be prompted for the path interactively.
Prints detailed error messages if validation fails, or confirms success.

### CODE REVIEW

File: data_schema_validation/test/unit_test.py [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import io
 import pandas as pd
 import pytest
-from data_schema_validation.src.input_verifier import verify_deeplabcut_csv, list_bodyparts
+from capstone.data_schema_validation.src.csv_verifier import verify_deeplabcut_csv, list_bodyparts
 
 # --------- Helpers ---------
```
Comment (madhuritha22) [inline data_schema_validation/test/unit_test.py:6]: It would be good to add more negative tests: empty lists, missing time_ranges when auto detection is off, wrong types inside nested dicts, etc. That ensures the validator really catches bad inputs.

File: data_schema_validation/src/json_verifier.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+                f"head must contain exactly {required_keys}, got {set(head.keys())}")
+        else:
+            for k in required_keys:
+                if not isinstance(head[k], str):
+                    errors.append(
+                        f"head.{k} must be a string, got {type(head[k]).__name__}")
+
+    return errors
+
+# =============== MAIN SCRIPT ===============
```
Comment (madhuritha22) [inline data_schema_validation/src/json_verifier.py:87]: We may want to add logic for dependent fields, e.g. if bulk_input = false, then bulk_input_path might not be needed, or if auto_find_time_ranges = true, then time_ranges can be optional.

Comment (Jung0219) [reply data_schema_validation/src/json_verifier.py:87]: This is a good point, but I don't think I can completely carve out each field because the structure is required by the file we're writing. I'll instead make the file raise a warning (suggestion) when file path is provided even when it's not needed.

File: data_schema_validation/test/unit_test.py (hunk 1 above)
Comment (Jung0219) [reply data_schema_validation/test/unit_test.py:6]: I added a unit test that checks if the input file is empty or not.

File: data_schema_validation/src/json_verifier.py [hunk 3]
```diff
@@ -56,16 +56,28 @@ def extra_checks(config: dict):
     # right_fin and left_fin must have at least 2 points
     for fin_key in ["right_fin", "left_fin"]:
         if fin_key in points:
-            if not isinstance(points[fin_key], list) or len(points[fin_key]) < 2:
+            fin_val = points[fin_key]
```
Comment (madhuritha22) [inline data_schema_validation/src/json_verifier.py:59]: Looks good to me now — the element-type check you added addresses the earlier concern. Thanks for making the update!

File: data_schema_validation/src/json_verifier.py (hunk 2 above)
Comment (madhuritha22) [reply data_schema_validation/src/json_verifier.py:120]: Thanks for adding the guidance_messages function. That’s exactly the kind of user-facing guidance I was hoping for. I agree it’s fine to keep the fields in the schema but surface a warning if they’re unnecessary — this makes it clear without breaking structure. Looks good to me.
